### PR TITLE
[vividus] Mark deprecated composite steps in printing steps task

### LINKED ISSUE #4048: Improve `printSteps` to print deprecation message for composite steps
Improve `printSteps` to print deprecation message for composite steps

_Originally posted by @valfirst in https://github.com/vividus-framework/vividus/pull/4027#discussion_r1230564110_

### PR DESCRIPTION
**Pull request by draker94**
Closes #4048

### CODE REVIEW

File: vividus/src/main/java/org/vividus/runner/DeprecatedStepsReplacer.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public final class DeprecatedStepsReplacer
     private static final String END_OF_STEP_LOOKAHEAD = "(?=\\h*(\\v|$))";
     private static final Pattern NESTED_STEPS_TABLE_PATTERN = Pattern.compile("(\\v|^).\\h*step\\h*.\\v");
     private static final Pattern COMPOSITE_DEPRECATION_NOTIFICATION
-            = Pattern.compile("!--\\s+WARNING: The step .* is deprecated and will be removed in .*");
+            = Pattern.compile("!--\\s+Deprecated: The step .* is deprecated and will be removed in .*");
```
Comment (uarlouski) [inline vividus/src/main/java/org/vividus/runner/DeprecatedStepsReplacer.java:78]: can we do this in upper case or case insensitive? it's just more visible this way

Comment (draker94) [reply vividus/src/main/java/org/vividus/runner/DeprecatedStepsReplacer.java]: > can we do this in upper case or case insensitive? it's just more visible this way

Added `CASE_INSENSITIVE` flags.

Comment (draker94) [reply vividus/src/main/java/org/vividus/runner/DeprecatedStepsReplacer.java]: > > can we do this in upper case or case insensitive? it's just more visible this way
> 
> Added `CASE_INSENSITIVE` flags.

But actually it doesn't make any sense to do  deprecation pattern case insensitive. 
I think in this place better to leave only the uppercase

File: docs/modules/developer-guides/pages/deprecating-steps.adoc [hunk 2]
```diff
@@ -59,13 +59,13 @@ org.vividus.log.LoggingStoryReporter - The step: "Given the step with parameters
 
 [NOTE]
 ====
-If composite step cannot be deprecated using formatter pattern it still must have the comment with the certain pattern for user notification in step replacer: `!--\s+WARNING: The step .* is deprecated and will be removed in .*`.
+If composite step cannot be deprecated using formatter pattern it still must have the comment with the certain pattern for user notification in step replacer: `!--\s+DEPRECATED: The step .* is deprecated and will be removed in .*`.
```
Comment (valfirst) [inline docs/modules/developer-guides/pages/deprecating-steps.adoc:62]: I guess you should update all composite steps to follow new pattern

File: vividus/src/main/java/org/vividus/runner/StepsPrinter.java [hunk 3]
```diff
@@ -49,6 +51,8 @@ public class StepsPrinter
     private static final String DEPRECATED = "DEPRECATED";
     private static final String COMPOSITE = "COMPOSITE IN STEPS FILE";
     private static final String EMPTY = "";
+    private static final Pattern DEPRECATION_COMMENT_PATTERN = Pattern.compile("^!--\\h+Deprecated:\\h+.*",
+            Pattern.CASE_INSENSITIVE);
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/runner/StepsPrinter.java:55]: I don't think it should be case insensitive, markers format should be strict: only upper case

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**